src/kademlia/rpc_manager.cpp

```diff
@@ -158,7 +159,7 @@ void observer::set_id(node_id const& id)
 enum { observer_size = max3<
 	sizeof(find_data_observer)
 	, sizeof(announce_observer)
-	, sizeof(null_observer)
+	, sizeof(put_data_observer)
```

Comment:
I think you need the null_observer in here too (for completeness). (extending max to take one more argument should be pretty straight-forward, or you could just combine multiple max3). or do you think there's no point?